cohttp-lwt/src/server.ml

```diff
@@ -111,32 +112,32 @@ module Make (IO : S.IO) = struct
                 `Response rsp))
       (fun () -> Body.drain_body body)
 
-  let handle_response ~keep_alive oc res body conn_closed handle_client =
+  type conn_action = Call_conn_closed | Call_conn_closed_and_drain of Body.t
+
+  let handle_response ~keep_alive oc res body _conn_closed handle_client =
     IO.catch (fun () ->
         let flush = Response.flush res in
         Response.write ~flush
           (fun writer -> Body.write_body (Response.write_body writer) body)
           res oc)
     >>= function
     | Ok () ->
-        if keep_alive then handle_client oc
-        else
-          let () = conn_closed () in
-          Lwt.return_unit
+        if keep_alive then handle_client oc else Lwt.return Call_conn_closed
     | Error e ->
         Log.info (fun m -> m "IO error while writing body: %a" IO.pp_error e);
-        conn_closed ();
-        Body.drain_body body
+        Lwt.return (Call_conn_closed_and_drain body)
 
   let rec handle_client ic oc conn spec =
+    let _, conn_id = conn in
     Request.read ic >>= function
     | `Eof ->
-        spec.conn_closed conn;
-        Lwt.return_unit
+        Log.debug (fun m ->
+            m "Got EOF while handling client: %s"
+              (Cohttp.Connection.to_string conn_id));
```

Comment:
```suggestion
              (Cohttp.Connection.to_string (snd conn)));
```

Reply:
Done in [fixup4](https://github.com/gabrielmoise17/ocaml-cohttp/pull/3/commits/4b6dc15371d36a4e6fdfe9b8962b46c12e1caf62)